cob_twist_controller/src/limiters/limiter.cpp

```diff
@@ -320,3 +356,40 @@ KDL::JntArray LimiterIndividualJointAccelerations::enforceLimits(const KDL::JntA
     return q_dot_norm;
 }
 /* END LimiterIndividualJointAccelerations **********************************************************************/
+
+/* BEGIN LimiterCartesianVelocities ********************************************************************/
+/**
+ * This implementation implements a saturation function to the Cartesian twists
+ */
+KDL::Twist LimiterCartesianVelocities::enforceCartesianLimits(const KDL::Twist& v_in) const
+{
+    KDL::Twist v_out(v_in);
+    double factor = 1.0;
+    // linear limts
+    for (unsigned int i = 0; i < 3; i++)
+    {
+            if (factor< v_out(i) / limiter_params_.max_lin_vel)
+            {
+            	v_out(i) = limiter_params_.max_lin_vel;
+            }
+            if (-factor > v_out(i) / limiter_params_.max_lin_vel)
+            {
+            	v_out(i) = -limiter_params_.max_lin_vel;
+            }
```

Comment:
you could use the functions from https://github.com/ipa320/cob_control/blob/indigo_dev/cob_frame_tracker/src/cob_frame_tracker.cpp#L284-L302